test/shared/src/main/scala/zio/test/TestAspect.scala

```diff
@@ -168,6 +168,35 @@ object TestAspect extends TimeoutVariants {
       }
     }
 
+  /**
+   * An aspect that makes a test succeed if it fails for any cause.
```

Comment:
I think we need to update this a bit because it makes a test that failed for any reason into a success, it doesn't have to be because the test failed with a `Cause`. It could have just failed with an `AssertionFailure`. Also clarify that it makes a previously successful test into a failure.